containers/src/Kokkos_DynRankView.hpp

```diff
@@ -580,9 +580,38 @@ class DynRankView : private View<DataType*******, Properties...> {
              index_type i6 = 0) const {
     return view_type::operator()(i0, i1, i2, i3, i4, i5, i6);
   }
+
+// This is an accomodation for Phalanx, who use the operator[] to access all
+// memory
+#ifdef KOKKOS_ENABLE_DEPRECATED_CODE_4
   KOKKOS_FUNCTION reference_type operator[](index_type i0) const {
+    if constexpr (std::is_same<typename drvtraits::value_type,
+                               typename drvtraits::scalar_array_type>::value) {
+      return view_type::data()[i0];
+    } else {
+      const size_t dim_scalar = view_type::impl_map().dimension_scalar();
+      const size_t bytes      = view_type::span() / dim_scalar;
+
+      using tmp_view_type = Kokkos::View<
+          DataType*, typename traits::array_layout,
+          typename traits::device_type,
+          Kokkos::MemoryTraits<traits::memory_traits::is_unmanaged |
+                               traits::memory_traits::is_random_access |
+                               traits::memory_traits::is_atomic>>;
```

Comment:
Maybe a dumb question, but aren't these all booleans? Thus the memory traits would either be 0 or Unmanaged?

Reply:
No, you probably want `MemoryTraitsFlags`
```suggestion
          Kokkos::MemoryTraits<Kokkos::Unmanaged |
                               Kokkos::RandomAccess |
                               Kokkos::Atomic>>;
```

Reply:
Would you want to take the values from traits, rather than enabling all three flags?

Reply:
Like
```
(traits::memory_traits::is_unmanaged ? Kokkos::Unmanaged : 0) | ...
```

Reply:
yeah but I'm honestly not sure what this function is supposed to do

Reply:
To me it looks like we want to reinterpret the DynRankView as 1-D View to access all elements preserving the memory traits (ignoring `Restrict` and `Aligned`).

Reply:
@crtrott just copied the old code over, so it looks like this might have been broken for the last 5 years? :P

Reply:
I mean it apparently worked good enough for `Trilinos`. 🙂